tests/conftest.py

```diff
@@ -50,16 +52,34 @@ def file_mediatypes_library():
     return lib
 
 
+@pytest.fixture(scope="session")
+def temp_dir():
+    """Creates a shared library path for tests, and cleans it up after the session."""
+    temp_root = Path(tempfile.gettempdir())
+    test_dir = temp_root / "ts-test-temp"
+
+    if test_dir.exists():
+        shutil.rmtree(test_dir)
+    test_dir.mkdir(parents=True)
+
+    yield test_dir
+
+    shutil.rmtree(test_dir, ignore_errors=True)
```

Comment:
I don't believe it's necessary to yield and remove the `test_dir` here, since the test(s) that rely on this path (in this case the `thumbs_path` and `test_path` code) recreates `test_dir` anyway via the `parents=True` argument in `mkdir`. And since the later code's paths are never cleaned up, the real logic doing the cleanup is always going to be the `if test_dir.exists()` check that precedes this.
I've tested this on Windows, macOS, and Linux and it works as intended. 
```suggestion
    return test_dir
```

Reply:
Removing the yield + rmtree() here should be fine 👍